src/components/smartSearch/EditTargetDialog.tsx

```diff
@@ -0,0 +1,132 @@
+import { FormattedMessage as Msg } from 'react-intl';
+import { useState } from 'react';
+import { Box, Button, ButtonBase, Card, CardContent, Dialog, DialogContent, IconButton, Typography } from '@material-ui/core';
+import { Delete, Edit } from '@material-ui/icons';
+
+import All from './filters/All';
+import MostActive from './filters/MostActive';
+import patchTaskItem from 'fetching/tasks/patchTaskItem';
+import { useRouter } from 'next/router';
+import { ZetkinSmartSearchFilter } from 'types/zetkin';
+import { useMutation, useQueryClient } from 'react-query';
+
+interface EditTargetDialogProps {
+    filterSpec: ZetkinSmartSearchFilter[];
+    onDialogClose: () => void;
+    open: boolean;
+}
+
+enum FILTER_TYPE {
+    ALL ='all',
+    MOST_ACTIVE ='most_active',
+}
```

Comment:
i think in `src/types` we can make a `smartSearch.ts` and put all types for this stuff in there